src/components/GitStash.tsx

```diff
@@ -0,0 +1,386 @@
+
+  /**
+   * message corresponding to the stash
+   */
+  message: string;
+
+  /**
+   * Wrap mouse event handler to stop event propagation
+   * @param fn Mouse event handler
+   * @returns Mouse event handler that stops event from propagating
+   */
+  stopPropagationWrapper: (
+    fn: React.EventHandler<React.MouseEvent>
+  ) => React.EventHandler<React.MouseEvent>;
+}
+
+/**
+ * Dropdown for each entry in the stash
+ */
+const GitStashEntry: React.FunctionComponent<IGitStashEntryProps> = (
+  props: IGitStashEntryProps
+) => {
+  const [showStashFiles, setShowStashFiles] = React.useState(false);
+
+  const nFiles = props?.files?.length;
+
+  // const gitStashPop = async (index: number): Promise<void> => {
+  //   try {
+  //     await props.model.stash_pop(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  // const gitStashDrop = async (index: number): Promise<void> => {
+  //   console.log('Call Git Stash Drop', index);
+  //   try {
+  //     await props.model.stash_drop(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  // const gitStashApply = async (index: number): Promise<void> => {
+  //   try {
+  //     await props.model.stash_apply(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  const getFilePath = (file: string) => {
+    // Root directory
+    if (!file.includes('/')) {
+      return {
+        name: file,
+        path: ''
+      };
+    } else {
+      // One folder
+      const fileSplit = file.split('/');
+      if (fileSplit.length == 2) {
+        return {
+          path: fileSplit[0],
+          name: fileSplit[1]
+        };
+        // More than one nested folder
+      } else {
+        return {
+          name: fileSplit.pop(),
+          path: fileSplit.join('/')
+        };
+      }
+    }
+  };
+
+  return (
+    <div className={sectionFileContainerStyle}>
+      <div
+        className={sectionAreaStyle}
+        onClick={() => {
+          if (props.collapsible && props?.files.length > 0) {
+            setShowStashFiles(!showStashFiles);
+          }
+        }}
+      >
+        {props.collapsible && (
+          <button className={changeStageButtonStyle}>
+            {showStashFiles && props?.files.length > 0 ? (
+              <caretDownIcon.react />
+            ) : (
+              <caretRightIcon.react />
+            )}
+          </button>
+        )}
+        <span
+          className={sectionHeaderLabelStyle}
+          style={{ display: 'flex', justifyContent: 'space-between' }}
+        >
+          <p>
+            {props.message} (on {props.branch})
+          </p>
+          <span style={{ display: 'flex' }}>{props.actions}</span>
+        </span>
+      </div>
+      {showStashFiles && (
+        <FixedSizeList
+          height={Math.max(
+            Math.min(props.height - HEADER_HEIGHT, nFiles * ITEM_HEIGHT),
+            ITEM_HEIGHT
+          )}
+          itemCount={nFiles}
```

Comment:
Let's make this a proper list:

```suggestion
          innerElementType="ul"
          itemCount={nFiles}
```

Reply:
I'm having trouble styling the ul from FixedSizeList..  I tried modifying the FixedSizeList class styles inline and in the GitStashStyle and can't seem to select it.

<img width="1204" alt="image" src="https://user-images.githubusercontent.com/94336773/230677115-e40b603b-cc80-4222-b5e3-f3703a64aa02.png">

Reply:
Nevermind, was able to fix this with @basokant 's help.